src/System.Net.HttpListener/tests/HttpListenerAuthenticationTest.cs

```diff
@@ -384,22 +390,21 @@ public void Realm_SetDisposed_ThrowsObjectDisposedException()
 
         public async Task<HttpResponseMessage> AuthenticationFailure(HttpClient client, HttpStatusCode errorCode)
         {
+            client.Timeout = new TimeSpan(0, 0, 1);
             Task<HttpResponseMessage> clientTask = client.GetAsync(_factory.ListeningUrl);
 
             // The server task will hang forever if it is not cancelled.
             var tokenSource = new CancellationTokenSource();
             Task<HttpListenerContext> serverTask = Task.Run(() => _listener.GetContext(), tokenSource.Token);
-
+            
             // The client task should complete first - the server should send a 401 response.
             Task resultTask = await Task.WhenAny(clientTask, serverTask);
             tokenSource.Cancel();
             if (resultTask == serverTask)
             {
-                await serverTask;
+                await clientTask;
```

Comment:
Why did you change this?  If auth fails, which it's supposed to, the serverTask should never complete, because GetContext won't recognize an auth failure; in that case for the WhenAny to complete, it must be the clientTask that completed.  If the WhenAny completes due to the serverTask completing, then something went very wrong, and we want to await the serverTask to propagate any exception it may have had.  We then assert that the client task was in fact the one returned from the WhenAny, so that if for some reason the serverTask was the one to complete but completed without error, we'll then fail in the assert.